ci/azure-pipelines-steps.yml

```diff
@@ -3,6 +3,24 @@ parameters:
   installer: none
 
 steps:
+- task: CacheBeta@0
+  displayName: pip cache
+  inputs:
+    key: pip | $(Agent.OS)
+    path: $(PIP_CACHE_DIR)
+
+- task: CacheBeta@0
+  displayName: HDF5 cache
+  inputs:
+    key: HDF5 | $(Agent.OS)
```

Comment:
I haven't looked into Azure caching, but should the cache key also include the HDF5 version?

Reply:
Also maybe the OS version? If we have tests on Ubuntu 16.04 and 18.04, HDF5 compiled on one wouldn't necessarily work on the other.

The bit I don't know is whether the cache is already specific to each separate pipeline - if it is, this doesn't matter.